src/main/java/hudson/plugins/s3/Uploads.java

```diff
@@ -21,7 +21,7 @@ private Uploads() {}
     private final transient HashMap<FilePath, Upload> startedUploads = new HashMap<>();
     private final transient HashMap<FilePath, InputStream> openedStreams = new HashMap<>();
 
-    public void startUploading(TransferManager manager, FilePath file, InputStream inputsStream, String bucketName, String objectName, ObjectMetadata metadata) throws AmazonServiceException {
+    public Upload startUploading(TransferManager manager, FilePath file, InputStream inputsStream, String bucketName, String objectName, ObjectMetadata metadata) throws AmazonServiceException {
```

Comment:
You don't to change the signature of this function. You know `FilePath file` and you know fact that this method puts `Upload` to the map. So, you can access `Upload` via this map in any moment.
Just create a new getter for that `getUploadByFilename`.
(I don't know Jenkins very well, but usually changing signatures is bad idea :) )